x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/index/mapper/GeoShapeWithDocValuesFieldMapper.java

```diff
@@ -164,16 +197,29 @@ public GeoShapeWithDocValuesFieldMapper build(MapperBuilderContext context) {
                 stored.get(),
                 orientation.get().value(),
                 parser,
+                scriptValues(),
                 geoFormatterFactory,
                 meta.get()
             );
+            if (script.get() == null) {
+                return new GeoShapeWithDocValuesFieldMapper(
```

Comment:
The base class expect a different constructor to be used. This might be a nice follow up but not for this PR as it touches other mappings like geo_point.